integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/ViewPopRedirectTests/RedirectTest.java

```diff
@@ -111,16 +104,48 @@ public void testRedirect() throws Exception {
         requestHeader.put("Accept", "application/json");
 
         HttpRequestUtil.doPost(new URL(getProxyServiceURLHttp(PROXY_NAME)), inputPayload, requestHeader);
-        Thread.sleep(5000); //Make sure that the Message Processor has time to deactivate
-        Assert.assertFalse(messageProcessorClient.isActive(PROCESSOR_NAME), "Message processor" + PROCESSOR_NAME + "should not be active, " +
-                "but it is active.");
-
+        Awaitility.await()
+                .pollInterval(4, TimeUnit.MILLISECONDS)
+                .atMost(2, TimeUnit.SECONDS)
```

Comment:
Please make this to 5 minutes  ( 300 seconds ) and polling as 30 **seconds** .

Please do the same in other places too.